awesome_cart/awesome_cart/doctype/awc_transaction/awc_transaction.py

```diff
@@ -21,13 +21,28 @@
 	"Debug": 3
 }
 
+def call_hook(hook_name, **kwargs):
+	hooks = frappe.get_hooks(hook_name) or []
+	for hook in hooks:
+		# don't allow hooks to break processing
+		try:
+			frappe.call(hook, **kwargs)
+		except Exception:
+			log("Error calling hook method: {}->{}".format(hook_name, hook))
```

Comment:
Man I'm not sure about this silently failing. Atleast Sentry should pick this up somehow.

Reply:
do we really want a hook to break a so after taking payments tho?
because that will happen if I let an exception from a hook bubble up.

Figured the happy medium would be to put it on the log. Is there a python or server side to sentry? maybe we could actively send these there on exceptions?

Reply:
Yeah of course, Sentry pushes errors via it's Python API 

https://docs.sentry.io/clients/python/usage/#capture-an-error

If you wanna test this, go to "Sentry Settings" in your local and put in the DSN for the Sandbox Project in our Sentry account. 

Note : We use the deprecated Python Client, so ignore the warning.